src/github.com/getlantern/balancer/balancer.go

```diff
@@ -102,11 +102,17 @@ func (b *Balancer) Dial(network, addr string) (net.Conn, error) {
 		trustedOnly = true
 	}
 
+	var lastDialer *dialer
 	for i := 0; i < dialAttempts; i++ {
 		d, pickErr := b.pickDialer(trustedOnly)
 		if pickErr != nil {
 			return nil, pickErr
 		}
+		if d == lastDialer {
```

Comment:
That said, I think this is clearly an improvement, and I'll go ahead and merge it. We can open other tickets if necessary, but what do you think about the above assessment? Also curious about your thoughts on this @oxtoacart. I'm basically worried about users in China, for example, getting unlucky when the balancer chooses to load some key css file from Amsterdam. That would definitely result in the page appearing slow, and much slower than in free!